src/main/java/be/seeseemelk/mockbukkit/entity/EnderSignalMock.java

```diff
@@ -0,0 +1,108 @@
+package be.seeseemelk.mockbukkit.entity;
+
+import be.seeseemelk.mockbukkit.ServerMock;
+import com.google.common.base.Preconditions;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.entity.EnderSignal;
+import org.bukkit.entity.EntityType;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.UUID;
+
+public class EnderSignalMock extends EntityMock implements EnderSignal
+{
+
+	private Location targetLocation = null;
+	private boolean surviveAfterDeath = true;
+	private ItemStack item = new ItemStack(Material.ENDER_EYE);
+	private int life = 0;
+
+	/**
+	 * Constructs a new EntityMock on the provided {@link ServerMock} with a specified {@link UUID}.
+	 *
+	 * @param server The server to create the entity on.
+	 * @param uuid   The UUID of the entity.
+	 */
+	public EnderSignalMock(@NotNull ServerMock server, @NotNull UUID uuid)
+	{
+		super(server, uuid);
+	}
+
+	@Override
+	public @NotNull Location getTargetLocation()
+	{
+		Preconditions.checkState(targetLocation != null, "Target location has not been set!");
+		return targetLocation;
+	}
+
+	@Override
+	public void setTargetLocation(@NotNull Location location)
+	{
+		this.setTargetLocation(location, true);
+	}
+
+	@Override
+	public void setTargetLocation(@NotNull Location location, boolean update)
+	{
+		Preconditions.checkArgument(location != null, "Location cannot be null");
+		Preconditions.checkArgument(this.getWorld().equals(location.getWorld()), "Cannot target EnderSignal across worlds");
+		this.targetLocation = location;
```

Comment:
Looking into the implementation, seems like only the block coordinates are stored. Maybe we should do a toBlockLocation before:
```suggestion
		this.targetLocation = location.toBlockLocation();
```